frontend/src/app/libs/common-components/components/input/input.component.ts

```diff
@@ -0,0 +1,285 @@
+import { ChangeDetectionStrategy, Component, ElementRef, forwardRef, Input, EventEmitter, Output, OnInit, Optional, Host, SkipSelf, ChangeDetectorRef, AfterViewInit } from '@angular/core';
+import { AsyncValidatorFn, ControlContainer, ControlValueAccessor, FormBuilder, FormControl, FormGroup, NG_VALUE_ACCESSOR, ValidationErrors, ValidatorFn} from '@angular/forms';
+import { Observable } from 'rxjs';
+import { fromPromise } from 'rxjs/internal/observable/fromPromise';
+import { take } from 'rxjs/operators';
+import { ComponentTheme } from '../../shared/component-theme.enum';
+import { InputType } from '../../shared/input-type.enum';
+
+@Component({
+	selector: 'app-input',
+	templateUrl: './input.component.html',
+	styleUrls: ['./input.component.scss'],
+	providers: [
+		{
+			provide: NG_VALUE_ACCESSOR,
+			useExisting: forwardRef(() => InputComponent),
+			multi: true
+		}
+	],
+	changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class InputComponent<T> implements ControlValueAccessor, OnInit, AfterViewInit {
+
+	private syncValidators: ValidatorFn[] | null = null;
+	private asyncValidators: AsyncValidatorFn[] | null = null;
+	private defaultValue: string | boolean;
+
+	@Input() public inputValidators: ValidatorFn[] = null;
+
+	@Input() public inputValidatorsAsync: AsyncValidatorFn[] = null;
+
+	@Input() public changeValidators: ValidatorFn[] = null;
+
+	@Input() public changeValidatorsAsync: AsyncValidatorFn[] = null;
+
+	@Input() public formControlName: string;
+
+	@Input() public disabled: boolean = false;
+
+	@Input() public formControl: FormControl;
+
+	@Input() public value: string | boolean = '';
```

Comment:
no need to set default value
also `:number | string | boolean | null | undefined`